src/ttl.ts

```diff
@@ -3,46 +3,123 @@ import { client } from './app';
 import { getServerMessageTtlMillis } from './sqlite';
 
 const lastDeletedMessages: Record<string, string> = {};
-const nonpermissibleChannels: Record<string, number> = {};
 
 export async function continuallyRetrieveMessages(): Promise<void> {
   while (true) {
     await retrieveMessages();
   }
 }
 
-async function handleDeletesForMessagesOlderThan14Days(
+async function retrieveMessages(): Promise<void> {
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  for (const item of client.channels.cache) {
+    const channel = item[1];
+    if (!channel.isText() || channel.type === 'DM') {
+      continue;
+    }
+    if (!canGetAndDeleteMessages(channel)) {
+      continue;
+    }
+    if (!lastDeletedMessages[channel.id]) {
+      lastDeletedMessages[channel.id] = channel.id;
+    }
+    try {
+      const messages = await channel.messages.fetch({ after: lastDeletedMessages[channel.id], limit: 100 });
+      const awaitedPromises = await handleDeletesForNonBulkDeletableMessages(channel.guildId, channel.id, messages);
+      if (awaitedPromises.length === 0) {
+        await handleDeletesForBulkDeletableMessages(channel.guildId, channel, messages);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+}
+
+/**
+ * This is a cache so we don't have to keep checking channel permissions all the time.
+ * Channel ID -> Timestamp of when it was added to the dict
+ */
+const channelsWithoutProperPermissions: { [channelId: string]: number } = {};
```

Comment:
As I mentioned on Discord, there is no need for a custom caching implementation as discord.js itself already caches channels and roles and thus permissions. Methods like `permissionsIn()` further below only access cache anyways. You'd notice if something actually does a network request if a) it's prefixed with `fetch` or b) if it returns a `Promise<>`.

Reply:
removed